storage/badger/transaction_results.go

```diff
@@ -6,52 +6,100 @@ import (
 	"github.com/dgraph-io/badger/v2"
 
 	"github.com/onflow/flow-go/model/flow"
+	"github.com/onflow/flow-go/module"
 	"github.com/onflow/flow-go/storage"
 	"github.com/onflow/flow-go/storage/badger/operation"
 )
 
 type TransactionResults struct {
-	db *badger.DB
+	db    *badger.DB
+	cache *Cache
 }
 
-func NewTransactionResults(db *badger.DB) *TransactionResults {
-	return &TransactionResults{
-		db: db,
-	}
+type BlockIDTransactionID struct {
+	BlockID       flow.Identifier
+	TransactionID flow.Identifier
+}
+
+func KeyFromBlockIDTransactionID(blockID flow.Identifier, txID flow.Identifier) string {
+	return fmt.Sprintf("%x%x", blockID, txID)
 }
 
-// Store will store the transaction result for the given block ID
-func (tr *TransactionResults) Store(blockID flow.Identifier, transactionResult *flow.TransactionResult) error {
-	err := operation.RetryOnConflict(tr.db.Update, operation.SkipDuplicates(operation.InsertTransactionResult(blockID, transactionResult)))
+func KeyToBlockIDTransactionID(key string) (flow.Identifier, flow.Identifier, error) {
```

Comment:
Can we let the key be an `[]byte` to avoid the back-and-forth and error handling needed for the string conversion?